### PR TITLE
[Workspace] Dashboard admin(groups/users) implementation

### PR DESCRIPTION
**Pull request by yubonluo**
### Description
This PR set the backend roles in groups or users, whoever has the backend roles or exactly match the user or groups ids defined in this yml.config/dynamic config will be regard as dashboard admin.
Dashboard admin will have the access to all the workspaces and objects inside OpenSearch Dashboards.

### Issues Resolved

https://github.com/opensearch-project/OpenSearch-Dashboards/issues/6389

## Screenshot

No UI change in this PR.

### Yml config test
1. Clone the latest code and clone security-dashboards-plugin to the plugins directory
2. Build the security-dashboards-plugin and bootstrap
3. Download the patch file [test.patch](https://github.com/opensearch-project/OpenSearch-Dashboards/files/14975232/test.patch) and yarn `git apply ./test.patch`
4. Modify `config/opensearch_dashboards.yml`
```yml
workspace.enabled: true 
opensearch.hosts: ["https://localhost:9200"]
opensearch.username: "username"
opensearch.password: "password" 
opensearch.ssl.verificationMode: none
savedObjects.permission.enabled: true
```
5. Two tests ways
   * **Yml config**
      1. Modify `config/opensearch_dashboards.yml`
      ```yml
      opensearchDashboards.dashboardAdmin.groups: ["dashboard_admin"]
      opensearchDashboards.dashboardAdmin.users: ["dashboard_admin"]
      ```
   * **Dynamic application config**
      1. Modify `config/opensearch_dashboards.yml`
      ```yml
      application_config.enabled: true
      ```
      2. Update the configuration of an entity by CURL command
      ```
      curl --insecure {osd endpoint}/api/appconfig/opensearchDashboards.dashboardAdmin.users -X POST -H 'Accept: application/json' -H 'Content-Type: application/json' -H 'osd-xsrf: osd-fetch' -H 'Sec-Fetch-Dest: empty' --data-raw '{"newValue":"[\"dashboard_admin\"]"}' -u 'username:password'
     ```
6. Run `yarn start --no-base-path` and visit http://localhost:5601
7. You can check whether it is dashboards admin in the console.
image

### Check List

- [x] All tests pass
  - [x] `yarn test:jest`
  - [x] `yarn test:jest_integration`
- [x] New functionality includes testing.
- [x] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/workspace/server/plugin.ts [hunk 1]
```diff
@@ -67,12 +77,51 @@ export class WorkspacePlugin implements Plugin<WorkspacePluginSetup, WorkspacePl
...
+      updateDashboardAdminStateForRequest(request, groups, users, configGroups, configUsers);
+      return toolkit.next();
+    });
+
+    this.workspaceSavedObjectsClientWrapper = new WorkspaceSavedObjectsClientWrapper(
+      this.permissionControl
+    );
+
+    core.savedObjects.addClientWrapper(
+      0,
```
Comment (Hailong-am) [inline src/plugins/workspace/server/plugin.ts:113]: we should use `PRIORITY_FOR_PERMISSION_CONTROL_WRAPPER` right?

File: src/plugins/workspace/server/saved_objects/workspace_saved_objects_client_wrapper.test.ts [hunk 2]
```diff
@@ -152,6 +161,21 @@ describe('WorkspaceSavedObjectsClientWrapper', () => {
         await wrapper.delete(...deleteArgs);
         expect(clientMock.delete).toHaveBeenCalledWith(...deleteArgs);
       });
+      it('should call client.delete if user/groups is dashboard admin', async () => {
```
Comment (Hailong-am) [inline src/plugins/workspace/server/saved_objects/workspace_saved_objects_client_wrapper.test.ts:164]:
```suggestion
      it('should bypass permission check for client.delete if user/groups is dashboard admin', async () => {
```

File: src/plugins/workspace/server/saved_objects/workspace_saved_objects_client_wrapper.test.ts [hunk 3]
```diff
@@ -152,6 +161,21 @@ describe('WorkspaceSavedObjectsClientWrapper', () => {
...
+          requestMock,
+        } = generateWorkspaceSavedObjectsClientWrapper(DASHBOARD_ADMIN);
+        expect(getWorkspaceState(requestMock)).toEqual({
+          isDashboardAdmin: true,
+        });
+        const deleteArgs = ['dashboard', 'not-permitted-dashboard'] as const;
+        await wrapper.delete(...deleteArgs);
+        expect(permissionControlMock.validate).not.toHaveBeenCalled();
+        expect(clientMock.delete).toHaveBeenCalledWith(...deleteArgs);
+      });
```
Comment (Hailong-am) [inline src/plugins/workspace/server/saved_objects/workspace_saved_objects_client_wrapper.test.ts:178]: can we combine those cases into one and verify wrapper is same as clientMock?

File: src/plugins/workspace/server/plugin.ts (hunk 1 above)
Comment (yubonluo) [reply src/plugins/workspace/server/plugin.ts:113]: Yes, I will fix it.

File: src/plugins/workspace/server/saved_objects/workspace_saved_objects_client_wrapper.test.ts (hunk 2 above)
Comment (yubonluo) [reply src/plugins/workspace/server/saved_objects/workspace_saved_objects_client_wrapper.test.ts:164]: Thanks, I will modify it in this way.

File: src/plugins/workspace/server/saved_objects/workspace_saved_objects_client_wrapper.test.ts (hunk 3 above)
Comment (yubonluo) [reply src/plugins/workspace/server/saved_objects/workspace_saved_objects_client_wrapper.test.ts]: Yes, I have combined these cases related to dashboard admin.

**Pull request closed without merging**